src/components/node-list/node-list-row.js

```diff
@@ -0,0 +1,85 @@
+import React from 'react';
+import classnames from 'classnames';
+import NodeIcon from '../icons/node-icon';
+import VisibleIcon from '../icons/visible';
+import InvisibleIcon from '../icons/invisible';
+
+const NodeListRow = ({
+  active,
+  checked,
+  children,
+  disabled,
+  id,
+  label,
+  name,
+  onMouseEnter,
+  onMouseLeave,
+  onChange,
+  onClick,
+  type
+}) => {
+  const VisibilityIcon = checked ? VisibleIcon : InvisibleIcon;
+
+  return (
+    <div
+      className={classnames('pipeline-nodelist__row kedro', {
+        'pipeline-nodelist__row--active': active,
+        'pipeline-nodelist__row--disabled': disabled
+      })}
+      onMouseEnter={onMouseEnter}
+      onMouseLeave={onMouseLeave}>
+      <button
+        onClick={onClick}
+        onFocus={onMouseEnter}
+        onBlur={onMouseLeave}
+        disabled={disabled}
+        className={classnames('pipeline-nodelist__row__text', {
+          'pipeline-nodelist__row__text--active': active,
+          'pipeline-nodelist__row__text--unchecked': !checked
+        })}
+        title={children ? null : name}>
+        <NodeIcon
+          className={classnames(
+            'pipeline-nodelist__row__type-icon pipeline-nodelist__row__icon',
+            {
+              'pipeline-nodelist__row__type-icon--unchecked': !checked,
+              'pipeline-nodelist__row__type-icon--nested': !children
+            }
+          )}
+          type={type}
+        />
+        <span
+          className={classnames('pipeline-nodelist__row__label', {
+            'pipeline-nodelist__row__label--active': active,
+            'pipeline-nodelist__row__label--disabled': disabled,
+            'pipeline-nodelist__row__label--unchecked': !checked
+          })}
+          dangerouslySetInnerHTML={{ __html: label }}
```

Comment:
@bru5 Hey I'm not sure I follow - how would a specialised component render it differently? Could you present an example please?

Reply:
I'm not sure I understand how the component would work other than like this:

```jsx
const RenderHTML = html => <span dangerouslySetInnerHTML={{ __html: html }} />


<span>
  <RenderHTML html={label} />
</span>
```

Reply:
I was just thinking more along the lines of `<NodeListLabel html={}>` just to keep this concern separate and ideally include some kind of comment there on why HTML is being supported.

But since it's not intended to be reusable outside of this context, it's probably fine as it is.

Reply:
Cool, yeah I think it'll be okay in this instance, but thanks for flagging!